File: week-2/homework/readability.c
#include <cs50.h>
#include <stdio.h>
// In order to use StrLen()
#include <string.h>
// Library from: https://stackoverflow.com/questions/8611815/determine-if-char-is-a-num-or-letter
#include <ctype.h>
// Library from: https://iq.opengenus.org/rounding-and-truncating-numbers-in-c/#:~:text=round()
#include <math.h>

int main(void)
{
    int letters = 0;
    int words = 0;
    int sentences = 0;

    string input = get_string("Text: ");
    int length = strlen(input);

    for (int i = 0; i < length; i++)
    {
        // Get the current character
        char c = input[i];
        // Check if alphabetical
        if (isalpha(c))
        {
            letters++;
        }
        // Check for spaces
        if (c == ' ')
        {
            words++;
        }
        // Check for punctuation
        if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }

    // Getting too high of a score because last word was being left off
    // this adds one to the final total to correct the formula below
    words = words + 1;

    // printf("%i - Letters \n %i - Words \n %i - Sentences \n", letters, words, sentences);

    // Coleman-Liau Index
    float L = ((float)letters / (float)words) * 100;
    float S = ((float)sentences / (float)words) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int indexRounded = round(index);
    if (indexRounded > 16)
    {
        printf("Grade 16+\n");
    }
    else if (indexRounded < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", indexRounded);
    }
}

File: week-6/homework/readability/readability.py
# TODO
# initialize variables
texts = input("Text: ")
alpha = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z']
letters = 0
words = 1
sentences = 0

# Iterate over index
for t in texts:
    if t.lower() in alpha:
        letters += 1
    elif t.isspace():
        words += 1
    elif t == "." or t == "?" or t == "!":
        sentences += 1
    else:
        pass

# calculation time
l = (letters / words) * 100
s = (sentences / words) * 100

index = 0.0588 * l - 0.296 * s - 15.8
index = round(index)

if index > 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

# debug
print("Letters: ", letters)
print("Words: ", words)
print("Sentences: ", sentences)